numba/cuda/cudadrv/libs.py

```diff
@@ -80,67 +79,97 @@ def _get_source_variable(lib, static=False):
         return get_cuda_paths()[dir_type].by
 
 
-def test(_platform=None, print_paths=True):
+def test():
     """Test library lookup.  Path info is printed to stdout.
     """
     failed = False
 
     # Check for the driver
     try:
         dlloader, candidates = locate_driver_and_loader()
-        locations = ", ".join(candidates)
-        print(f'Finding driver from candidates: {locations}...')
+        print('Finding driver from candidates:')
+        for location in candidates:
+            print(f'\t{location}')
         print(f'Using loader {dlloader}')
-        print('\ttrying to load driver', end='...')
+        print('\tTrying to load driver', end='...')
         dll, path = load_driver(dlloader, candidates)
-        print(f'\tok, loaded from {path}')
+        print('\tok')
+        print(f'\t\tLoaded from {path}')
     except CudaSupportError as e:
         print(f'\tERROR: failed to open driver: {e}')
         failed = True
 
+    # Find the absolute location of the driver on Linux. Various driver-related
+    # issues have been reported by WSL2 users, and it is almost always due to a
+    # Linux (i.e. not- WSL2) driver being installed in a WSL2 system.
+    # Providing the absolute location of the driver indicates its version
+    # number in the soname (e.g. "libcuda.so.530.30.02"), which can be used to
+    # look up whether the driver was intended for "native" Linux.
+    if sys.platform == 'linux' and not failed:
+        pid = os.getpid()
+        mapsfile = f'/proc/{pid}/maps'
+        try:
+            with open(mapsfile) as f:
+                maps = f.read()
+        # It's difficult to predict all that might go wrong reading the maps
+        # file - in case various error conditions ensue (the file is not found,
+        # not readable, etc.) we use OSError to hopefully catch any of them.
```

Comment:
Another option, if this approach does end up to be more problematic than anticipated, would perhaps be to use `dl_iterate_phdr`.